### PR TITLE
Update `r/azurerm_(linux|windows)_virtual_machine_scale_set`: add support for automatic instance repair

### PR DESCRIPTION
**Pull request by ArcturusZhang**
This PR introduces the automatic instance repairs policy to resource `azurerm_linux_virtual_machine_scale_set` and `azurerm_windows_virtual_machine_scale_set`.

Automatic instance repairs policy is currently in public preview, you can find more information about this by this [doc](https://docs.microsoft.com/en-us/azure/virtual-machine-scale-sets/virtual-machine-scale-sets-automatic-instance-repairs)

### CODE REVIEW

File: azurerm/internal/services/compute/tests/resource_arm_linux_virtual_machine_scale_set_other_test.go [hunk 1]
```diff
@@ -460,6 +460,38 @@ func TestAccAzureRMLinuxVirtualMachineScaleSet_otherTerminateNotification(t *tes
...
+			{
+				Config: testAccAzureRMLinuxVirtualMachineScaleSet_otherAutomaticRepairsPolicy(data, false),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMLinuxVirtualMachineScaleSetExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(
+				"admin_password",
+			),
+		},
```
Comment (katbyte) [inline azurerm/internal/services/compute/tests/resource_arm_linux_virtual_machine_scale_set_other_test.go:491]: could we add another step taking this back to on?

File: azurerm/internal/services/compute/tests/resource_arm_windows_virtual_machine_scale_set_other_test.go [hunk 2]
```diff
@@ -630,6 +630,38 @@ func TestAccAzureRMWindowsVirtualMachineScaleSet_otherTerminateNotification(t *t
...
+			{
+				Config: testAccAzureRMWindowsVirtualMachineScaleSet_otherAutomaticRepairsPolicy(data, false),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMLinuxVirtualMachineScaleSetExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(
+				"admin_password",
+			),
+		},
```
Comment (katbyte) [inline azurerm/internal/services/compute/tests/resource_arm_windows_virtual_machine_scale_set_other_test.go:661]: could we add a step taking this back to on?

File: azurerm/internal/services/compute/tests/resource_arm_linux_virtual_machine_scale_set_other_test.go (hunk 1 above)
Comment (ArcturusZhang) [reply azurerm/internal/services/compute/tests/resource_arm_linux_virtual_machine_scale_set_other_test.go:491]: Done

File: azurerm/internal/services/compute/tests/resource_arm_windows_virtual_machine_scale_set_other_test.go (hunk 2 above)
Comment (ArcturusZhang) [reply azurerm/internal/services/compute/tests/resource_arm_windows_virtual_machine_scale_set_other_test.go:661]: Done

**Comment by ArcturusZhang** [comment]
Hi @katbyte I have added another step that turns the automatic instance repair on again.
And I spotted I did not follow the step of `on -> off -> on` in the terminate notification feature, therefore I also included same changes to the terminate notification tests.

**Pull request merged**